Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/StepDefinitions/TagsStepDefinitions.cs

```diff
@@ -10,22 +11,52 @@ public class TagsStepDefinitions
 {
     private readonly ResponseContext _responseContext;
     private readonly ClientPool _clientPool;
+    private readonly TagsContext _tagsContext;
 
-    private ApiResponse<ListTagsResponse>? _listTagsResponse;
+    private CachedApiResponse<ListTagsResponse>? _listTagsResponse;
 
     public TagsStepDefinitions(ResponseContext responseContext, ClientPool clientPool)
     {
         _responseContext = responseContext;
         _clientPool = clientPool;
+        _tagsContext = new TagsContext();
```

Comment:
Even though we will probably scratch the context: next time you need to use one, remember to not create it in the step definitions. Instead, let the DI container inject it for you. Otherwise, it cannot be shared between step definitions, which is the main purpose of the contexts.